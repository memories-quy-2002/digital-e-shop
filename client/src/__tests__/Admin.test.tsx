import { render } from "@testing-library/react";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import AdminDashboard from "../components/pages/admin/AdminDashboard";
import ToastProvider from "../context/ToastContext";
import AdminProductPage from "../components/pages/admin/AdminProductPage";
import AdminOrderPage from "../components/pages/admin/AdminOrderPage";
import AdminCustomerPage from "../components/pages/admin/AdminAccountPage";
import AdminAddProductPage from "../components/pages/admin/AdminAddProductPage";

describe("AdminDashboard", () => {
    it("matches the AdminDashboard snapshot", async () => {
        const { asFragment } = render(
            <ToastProvider>
                <MemoryRouter initialEntries={["/admin"]}>
                    <Routes>
                        <Route path="/admin" element={<AdminDashboard />} />
                        <Route path="/admin/products" element={<AdminProductPage />} />
                        <Route path="/admin/orders" element={<AdminOrderPage />} />
                        <Route path="/admin/customers" element={<AdminCustomerPage />} />
                        <Route path="/admin/add" element={<AdminAddProductPage />} />
                    </Routes>
                    <AdminDashboard />
                </MemoryRouter>
            </ToastProvider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
