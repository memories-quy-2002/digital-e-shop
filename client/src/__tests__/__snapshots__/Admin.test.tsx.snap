// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminDashboard matches the AdminDashboard snapshot 1`] = `
<DocumentFragment>
  <div
    class="admin"
  >
    <div
      class="admin__layout"
    >
      <div
        class="admin__layout__sidebar"
      >
        <div
          class="admin__layout__sidebar__title"
        >
          <h3>
            DIGITAL-E
          </h3>
          <strong>
            Admin Panel
          </strong>
        </div>
        <div
          class="admin__layout__sidebar__information"
        >
          <img
            alt="admin_avatar"
            class="admin__layout__sidebar__information__img"
            src="product_placeholder.jpg"
          />
          <div
            class="admin__layout__sidebar__information__user"
          >
            <strong>
              Anonymous
            </strong>
            <span>
              Anonymous
            </span>
          </div>
        </div>
        <div
          class="admin__layout__sidebar__navigation"
        >
          <button
            class="active"
            type="button"
          >
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M924.8 385.6a446.7 446.7 0 0 0-96-142.4 446.7 446.7 0 0 0-142.4-96C631.1 123.8 572.5 112 512 112s-119.1 11.8-174.4 35.2a446.7 446.7 0 0 0-142.4 96 446.7 446.7 0 0 0-96 142.4C75.8 440.9 64 499.5 64 560c0 132.7 58.3 257.7 159.9 343.1l1.7 1.4c5.8 4.8 13.1 7.5 20.6 7.5h531.7c7.5 0 14.8-2.7 20.6-7.5l1.7-1.4C901.7 817.7 960 692.7 960 560c0-60.5-11.9-119.1-35.2-174.4zM482 232c0-4.4 3.6-8 8-8h44c4.4 0 8 3.6 8 8v80c0 4.4-3.6 8-8 8h-44c-4.4 0-8-3.6-8-8v-80zM270 582c0 4.4-3.6 8-8 8h-80c-4.4 0-8-3.6-8-8v-44c0-4.4 3.6-8 8-8h80c4.4 0 8 3.6 8 8v44zm90.7-204.5l-31.1 31.1a8.03 8.03 0 0 1-11.3 0L261.7 352a8.03 8.03 0 0 1 0-11.3l31.1-31.1c3.1-3.1 8.2-3.1 11.3 0l56.6 56.6c3.1 3.1 3.1 8.2 0 11.3zm291.1 83.6l-84.5 84.5c5 18.7.2 39.4-14.5 54.1a55.95 55.95 0 0 1-79.2 0 55.95 55.95 0 0 1 0-79.2 55.87 55.87 0 0 1 54.1-14.5l84.5-84.5c3.1-3.1 8.2-3.1 11.3 0l28.3 28.3c3.1 3.1 3.1 8.1 0 11.3zm43-52.4l-31.1-31.1a8.03 8.03 0 0 1 0-11.3l56.6-56.6c3.1-3.1 8.2-3.1 11.3 0l31.1 31.1c3.1 3.1 3.1 8.2 0 11.3l-56.6 56.6a8.03 8.03 0 0 1-11.3 0zM846 582c0 4.4-3.6 8-8 8h-80c-4.4 0-8-3.6-8-8v-44c0-4.4 3.6-8 8-8h80c4.4 0 8 3.6 8 8v44z"
              />
            </svg>
            Dashboard
          </button>
          <button
            class=""
            type="button"
          >
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M50.7 58.5L0 160H208V32H93.7C75.5 32 58.9 42.3 50.7 58.5zM240 160H448L397.3 58.5C389.1 42.3 372.5 32 354.3 32H240V160zm208 32H0V416c0 35.3 28.7 64 64 64H384c35.3 0 64-28.7 64-64V192z"
              />
            </svg>
            Products
          </button>
          <button
            class=""
            type="button"
          >
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 24C0 10.7 10.7 0 24 0H69.5c22 0 41.5 12.8 50.6 32h411c26.3 0 45.5 25 38.6 50.4l-41 152.3c-8.5 31.4-37 53.3-69.5 53.3H170.7l5.4 28.5c2.2 11.3 12.1 19.5 23.6 19.5H488c13.3 0 24 10.7 24 24s-10.7 24-24 24H199.7c-34.6 0-64.3-24.6-70.7-58.5L77.4 54.5c-.7-3.8-4-6.5-7.9-6.5H24C10.7 48 0 37.3 0 24zM128 464a48 48 0 1 1 96 0 48 48 0 1 1 -96 0zm336-48a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
              />
            </svg>
            Orders
          </button>
          <button
            class=""
            type="button"
          >
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
              />
            </svg>
            Customers
          </button>
        </div>
      </div>
      <div
        class="admin__layout__main"
      >
        <div
          class="admin__layout__main__header"
        >
          F
          <div
            style="display: flex; flex-direction: row; gap: 2rem; justify-content: center;"
          >
            <div>
              <button
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
                  />
                </svg>
                Home
              </button>
            </div>
            <div
              class="admin__layout__main__header__search"
            >
              <input
                placeholder="Search..."
                type="text"
              />
              <button
                type="submit"
              >
                <svg
                  fill="currentColor"
                  height="20"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M456.69 421.39 362.6 327.3a173.81 173.81 0 0 0 34.84-104.58C397.44 126.38 319.06 48 222.72 48S48 126.38 48 222.72s78.38 174.72 174.72 174.72A173.81 173.81 0 0 0 327.3 362.6l94.09 94.09a25 25 0 0 0 35.3-35.3zM97.92 222.72a124.8 124.8 0 1 1 124.8 124.8 124.95 124.95 0 0 1-124.8-124.8z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            style="display: flex; flex-direction: row; gap: 0.5rem; justify-content: center; align-items: center;"
          >
            <div
              class="admin__layout__main__header__notifications"
            >
              <svg
                class="admin__layout__main__header__notifications__icon"
                fill="currentColor"
                height="28"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="28"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M440.08 341.31c-1.66-2-3.29-4-4.89-5.93-22-26.61-35.31-42.67-35.31-118 0-39-9.33-71-27.72-95-13.56-17.73-31.89-31.18-56.05-41.12a3 3 0 0 1-.82-.67C306.6 51.49 282.82 32 256 32s-50.59 19.49-59.28 48.56a3.13 3.13 0 0 1-.81.65c-56.38 23.21-83.78 67.74-83.78 136.14 0 75.36-13.29 91.42-35.31 118-1.6 1.93-3.23 3.89-4.89 5.93a35.16 35.16 0 0 0-4.65 37.62c6.17 13 19.32 21.07 34.33 21.07H410.5c14.94 0 28-8.06 34.19-21a35.17 35.17 0 0 0-4.61-37.66zM256 480a80.06 80.06 0 0 0 70.44-42.13 4 4 0 0 0-3.54-5.87H189.12a4 4 0 0 0-3.55 5.87A80.06 80.06 0 0 0 256 480z"
                />
              </svg>
              <span
                class="admin__layout__main__header__notifications__badge"
              >
                1
              </span>
            </div>
            <button>
              LOGOUT
              <svg
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M304 336v40a40 40 0 0 1-40 40H104a40 40 0 0 1-40-40V136a40 40 0 0 1 40-40h152c22.09 0 48 17.91 48 40v40m64 160 80-80-80-80m-192 80h256"
                  fill="none"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="32"
                />
              </svg>
            </button>
          </div>
        </div>
         
        <div
          class="admin__dashboard"
        >
          <div
            class="admin__dashboard-header"
          >
            <h2>
              Dashboard
            </h2>
            <button
              class="btn btn-dark"
            >
              Download Report
            </button>
          </div>
          <div
            class="admin__dashboard-cards"
          >
            <div
              class="admin__dashboard-card"
              style="background-color: purple;"
            >
              <div
                class="admin__dashboard-card__title"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
                  />
                </svg>
                <strong>
                  Sales
                </strong>
              </div>
              <div
                class="admin__dashboard-card__details"
              >
                <p>
                  0
                </p>
                <span>
                  Total for this month 
                </span>
                0
              </div>
            </div>
            <div
              class="admin__dashboard-card"
              style="background-color: darkblue;"
            >
              <div
                class="admin__dashboard-card__title"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 640 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M608 64H32C14.33 64 0 78.33 0 96v320c0 17.67 14.33 32 32 32h576c17.67 0 32-14.33 32-32V96c0-17.67-14.33-32-32-32zM48 400v-64c35.35 0 64 28.65 64 64H48zm0-224v-64h64c0 35.35-28.65 64-64 64zm272 176c-44.19 0-80-42.99-80-96 0-53.02 35.82-96 80-96s80 42.98 80 96c0 53.03-35.83 96-80 96zm272 48h-64c0-35.35 28.65-64 64-64v64zm0-224c-35.35 0-64-28.65-64-64h64v64z"
                  />
                </svg>
                <strong>
                  Revenue
                </strong>
              </div>
              <div
                class="admin__dashboard-card__details"
              >
                <p>
                  $0.00
                </p>
                <span>
                  Total for this month 
                </span>
                0
              </div>
            </div>
            <div
              class="admin__dashboard-card"
              style="background-color: green;"
            >
              <div
                class="admin__dashboard-card__title"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M509.5 184.6L458.9 32.8C452.4 13.2 434.1 0 413.4 0H272v192h238.7c-.4-2.5-.4-5-1.2-7.4zM240 0H98.6c-20.7 0-39 13.2-45.5 32.8L2.5 184.6c-.8 2.4-.8 4.9-1.2 7.4H240V0zM0 224v240c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V224H0z"
                  />
                </svg>
                <strong>
                  Products
                </strong>
              </div>
              <div
                class="admin__dashboard-card__details"
              >
                <p>
                  0
                </p>
                <span>
                  Total 
                </span>
              </div>
            </div>
            <div
              class="admin__dashboard-card"
              style="background-color: blue;"
            >
              <div
                class="admin__dashboard-card__title"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                  />
                </svg>
                <strong>
                  Users
                </strong>
              </div>
              <div
                class="admin__dashboard-card__details"
              >
                <p>
                  0
                </p>
                <span>
                  Total 
                </span>
              </div>
            </div>
          </div>
          <div
            class="admin__dashboard-chart"
          >
            <div
              style="flex: 1;"
            >
              <h3>
                Sales Over Time
              </h3>
              <div
                class="recharts-responsive-container"
                style="width: 100%; height: 300px; min-width: 0;"
              />
            </div>
            <div
              style="flex: 1;"
            >
              <h3>
                Revenue Over Time
              </h3>
              <div
                class="recharts-responsive-container"
                style="width: 100%; height: 300px; min-width: 0;"
              />
            </div>
          </div>
          <div
            class="admin__dashboard-table"
          >
            <h3>
              Top-selling products
            </h3>
            <div
              class="table-responsive"
            >
              <table
                class="table table-striped table-borderless table-hover"
              >
                <thead>
                  <tr>
                    <th>
                      #
                    </th>
                    <th>
                      Name
                    </th>
                    <th>
                      Sales
                    </th>
                    <th>
                      Revenue
                    </th>
                  </tr>
                </thead>
                <tbody />
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="admin"
  >
    <div
      class="admin__layout"
    >
      <div
        class="admin__layout__sidebar"
      >
        <div
          class="admin__layout__sidebar__title"
        >
          <h3>
            DIGITAL-E
          </h3>
          <strong>
            Admin Panel
          </strong>
        </div>
        <div
          class="admin__layout__sidebar__information"
        >
          <img
            alt="admin_avatar"
            class="admin__layout__sidebar__information__img"
            src="product_placeholder.jpg"
          />
          <div
            class="admin__layout__sidebar__information__user"
          >
            <strong>
              Anonymous
            </strong>
            <span>
              Anonymous
            </span>
          </div>
        </div>
        <div
          class="admin__layout__sidebar__navigation"
        >
          <button
            class="active"
            type="button"
          >
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M924.8 385.6a446.7 446.7 0 0 0-96-142.4 446.7 446.7 0 0 0-142.4-96C631.1 123.8 572.5 112 512 112s-119.1 11.8-174.4 35.2a446.7 446.7 0 0 0-142.4 96 446.7 446.7 0 0 0-96 142.4C75.8 440.9 64 499.5 64 560c0 132.7 58.3 257.7 159.9 343.1l1.7 1.4c5.8 4.8 13.1 7.5 20.6 7.5h531.7c7.5 0 14.8-2.7 20.6-7.5l1.7-1.4C901.7 817.7 960 692.7 960 560c0-60.5-11.9-119.1-35.2-174.4zM482 232c0-4.4 3.6-8 8-8h44c4.4 0 8 3.6 8 8v80c0 4.4-3.6 8-8 8h-44c-4.4 0-8-3.6-8-8v-80zM270 582c0 4.4-3.6 8-8 8h-80c-4.4 0-8-3.6-8-8v-44c0-4.4 3.6-8 8-8h80c4.4 0 8 3.6 8 8v44zm90.7-204.5l-31.1 31.1a8.03 8.03 0 0 1-11.3 0L261.7 352a8.03 8.03 0 0 1 0-11.3l31.1-31.1c3.1-3.1 8.2-3.1 11.3 0l56.6 56.6c3.1 3.1 3.1 8.2 0 11.3zm291.1 83.6l-84.5 84.5c5 18.7.2 39.4-14.5 54.1a55.95 55.95 0 0 1-79.2 0 55.95 55.95 0 0 1 0-79.2 55.87 55.87 0 0 1 54.1-14.5l84.5-84.5c3.1-3.1 8.2-3.1 11.3 0l28.3 28.3c3.1 3.1 3.1 8.1 0 11.3zm43-52.4l-31.1-31.1a8.03 8.03 0 0 1 0-11.3l56.6-56.6c3.1-3.1 8.2-3.1 11.3 0l31.1 31.1c3.1 3.1 3.1 8.2 0 11.3l-56.6 56.6a8.03 8.03 0 0 1-11.3 0zM846 582c0 4.4-3.6 8-8 8h-80c-4.4 0-8-3.6-8-8v-44c0-4.4 3.6-8 8-8h80c4.4 0 8 3.6 8 8v44z"
              />
            </svg>
            Dashboard
          </button>
          <button
            class=""
            type="button"
          >
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M50.7 58.5L0 160H208V32H93.7C75.5 32 58.9 42.3 50.7 58.5zM240 160H448L397.3 58.5C389.1 42.3 372.5 32 354.3 32H240V160zm208 32H0V416c0 35.3 28.7 64 64 64H384c35.3 0 64-28.7 64-64V192z"
              />
            </svg>
            Products
          </button>
          <button
            class=""
            type="button"
          >
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 24C0 10.7 10.7 0 24 0H69.5c22 0 41.5 12.8 50.6 32h411c26.3 0 45.5 25 38.6 50.4l-41 152.3c-8.5 31.4-37 53.3-69.5 53.3H170.7l5.4 28.5c2.2 11.3 12.1 19.5 23.6 19.5H488c13.3 0 24 10.7 24 24s-10.7 24-24 24H199.7c-34.6 0-64.3-24.6-70.7-58.5L77.4 54.5c-.7-3.8-4-6.5-7.9-6.5H24C10.7 48 0 37.3 0 24zM128 464a48 48 0 1 1 96 0 48 48 0 1 1 -96 0zm336-48a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
              />
            </svg>
            Orders
          </button>
          <button
            class=""
            type="button"
          >
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
              />
            </svg>
            Customers
          </button>
        </div>
      </div>
      <div
        class="admin__layout__main"
      >
        <div
          class="admin__layout__main__header"
        >
          F
          <div
            style="display: flex; flex-direction: row; gap: 2rem; justify-content: center;"
          >
            <div>
              <button
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
                  />
                </svg>
                Home
              </button>
            </div>
            <div
              class="admin__layout__main__header__search"
            >
              <input
                placeholder="Search..."
                type="text"
              />
              <button
                type="submit"
              >
                <svg
                  fill="currentColor"
                  height="20"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M456.69 421.39 362.6 327.3a173.81 173.81 0 0 0 34.84-104.58C397.44 126.38 319.06 48 222.72 48S48 126.38 48 222.72s78.38 174.72 174.72 174.72A173.81 173.81 0 0 0 327.3 362.6l94.09 94.09a25 25 0 0 0 35.3-35.3zM97.92 222.72a124.8 124.8 0 1 1 124.8 124.8 124.95 124.95 0 0 1-124.8-124.8z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            style="display: flex; flex-direction: row; gap: 0.5rem; justify-content: center; align-items: center;"
          >
            <div
              class="admin__layout__main__header__notifications"
            >
              <svg
                class="admin__layout__main__header__notifications__icon"
                fill="currentColor"
                height="28"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="28"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M440.08 341.31c-1.66-2-3.29-4-4.89-5.93-22-26.61-35.31-42.67-35.31-118 0-39-9.33-71-27.72-95-13.56-17.73-31.89-31.18-56.05-41.12a3 3 0 0 1-.82-.67C306.6 51.49 282.82 32 256 32s-50.59 19.49-59.28 48.56a3.13 3.13 0 0 1-.81.65c-56.38 23.21-83.78 67.74-83.78 136.14 0 75.36-13.29 91.42-35.31 118-1.6 1.93-3.23 3.89-4.89 5.93a35.16 35.16 0 0 0-4.65 37.62c6.17 13 19.32 21.07 34.33 21.07H410.5c14.94 0 28-8.06 34.19-21a35.17 35.17 0 0 0-4.61-37.66zM256 480a80.06 80.06 0 0 0 70.44-42.13 4 4 0 0 0-3.54-5.87H189.12a4 4 0 0 0-3.55 5.87A80.06 80.06 0 0 0 256 480z"
                />
              </svg>
              <span
                class="admin__layout__main__header__notifications__badge"
              >
                1
              </span>
            </div>
            <button>
              LOGOUT
              <svg
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M304 336v40a40 40 0 0 1-40 40H104a40 40 0 0 1-40-40V136a40 40 0 0 1 40-40h152c22.09 0 48 17.91 48 40v40m64 160 80-80-80-80m-192 80h256"
                  fill="none"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="32"
                />
              </svg>
            </button>
          </div>
        </div>
         
        <div
          class="admin__dashboard"
        >
          <div
            class="admin__dashboard-header"
          >
            <h2>
              Dashboard
            </h2>
            <button
              class="btn btn-dark"
            >
              Download Report
            </button>
          </div>
          <div
            class="admin__dashboard-cards"
          >
            <div
              class="admin__dashboard-card"
              style="background-color: purple;"
            >
              <div
                class="admin__dashboard-card__title"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
                  />
                </svg>
                <strong>
                  Sales
                </strong>
              </div>
              <div
                class="admin__dashboard-card__details"
              >
                <p>
                  0
                </p>
                <span>
                  Total for this month 
                </span>
                0
              </div>
            </div>
            <div
              class="admin__dashboard-card"
              style="background-color: darkblue;"
            >
              <div
                class="admin__dashboard-card__title"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 640 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M608 64H32C14.33 64 0 78.33 0 96v320c0 17.67 14.33 32 32 32h576c17.67 0 32-14.33 32-32V96c0-17.67-14.33-32-32-32zM48 400v-64c35.35 0 64 28.65 64 64H48zm0-224v-64h64c0 35.35-28.65 64-64 64zm272 176c-44.19 0-80-42.99-80-96 0-53.02 35.82-96 80-96s80 42.98 80 96c0 53.03-35.83 96-80 96zm272 48h-64c0-35.35 28.65-64 64-64v64zm0-224c-35.35 0-64-28.65-64-64h64v64z"
                  />
                </svg>
                <strong>
                  Revenue
                </strong>
              </div>
              <div
                class="admin__dashboard-card__details"
              >
                <p>
                  $0.00
                </p>
                <span>
                  Total for this month 
                </span>
                0
              </div>
            </div>
            <div
              class="admin__dashboard-card"
              style="background-color: green;"
            >
              <div
                class="admin__dashboard-card__title"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M509.5 184.6L458.9 32.8C452.4 13.2 434.1 0 413.4 0H272v192h238.7c-.4-2.5-.4-5-1.2-7.4zM240 0H98.6c-20.7 0-39 13.2-45.5 32.8L2.5 184.6c-.8 2.4-.8 4.9-1.2 7.4H240V0zM0 224v240c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V224H0z"
                  />
                </svg>
                <strong>
                  Products
                </strong>
              </div>
              <div
                class="admin__dashboard-card__details"
              >
                <p>
                  0
                </p>
                <span>
                  Total 
                </span>
              </div>
            </div>
            <div
              class="admin__dashboard-card"
              style="background-color: blue;"
            >
              <div
                class="admin__dashboard-card__title"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                  />
                </svg>
                <strong>
                  Users
                </strong>
              </div>
              <div
                class="admin__dashboard-card__details"
              >
                <p>
                  0
                </p>
                <span>
                  Total 
                </span>
              </div>
            </div>
          </div>
          <div
            class="admin__dashboard-chart"
          >
            <div
              style="flex: 1;"
            >
              <h3>
                Sales Over Time
              </h3>
              <div
                class="recharts-responsive-container"
                style="width: 100%; height: 300px; min-width: 0;"
              />
            </div>
            <div
              style="flex: 1;"
            >
              <h3>
                Revenue Over Time
              </h3>
              <div
                class="recharts-responsive-container"
                style="width: 100%; height: 300px; min-width: 0;"
              />
            </div>
          </div>
          <div
            class="admin__dashboard-table"
          >
            <h3>
              Top-selling products
            </h3>
            <div
              class="table-responsive"
            >
              <table
                class="table table-striped table-borderless table-hover"
              >
                <thead>
                  <tr>
                    <th>
                      #
                    </th>
                    <th>
                      Name
                    </th>
                    <th>
                      Sales
                    </th>
                    <th>
                      Revenue
                    </th>
                  </tr>
                </thead>
                <tbody />
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="toast-container bottom-0 end-0 p-3"
    style="z-index: 1; position: fixed; bottom: 0px; right: 0px;"
  />
</DocumentFragment>
`;
